### PR TITLE
Add libtinyiiod package

### PR DESCRIPTION
**Pull request by tfcollins**
Add package for libtinyiiod library which is used by a number of projects (to be added)

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tfcollins** [comment]
@conda-forge/help-c-cpp ready for review!

File: recipes/libtinyiiod/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/analogdevicesinc/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 0b9bc22b2e0090b905fa369fcc652fe3522a68b0aad5ded1d30a1464058596d2
+
+build:
+  number: 0
```
Comment (carterbox) [inline recipes/libtinyiiod/meta.yaml:13]:
```suggestion
build:
  number: 0
  run_exports:
    - {{ pin_subpackage('libtinyiuod' max_pin='x.x.x') }}
```

Please add a run export which provides the correct version for ABI compatiblility with downstream packages.

File: recipes/libtinyiiod/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    # verify that headers get installed
+    - test -f $PREFIX/include/tinyiiod/tinyiiod.h  # [not win]
+    - test -f $PREFIX/include/tinyiiod/compat.h  # [not win]
+    - if not exist %PREFIX%\\Library\\include\tinyiiod\tinyiiod.h exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\include\tinyiiod\compat.h exit 1  # [win]
+
+    # verify that libraries get installed
+    - test -f $PREFIX/lib/tinyiiod${SHLIB_EXT}  # [not win]
+    - if not exist %PREFIX%\\Library\\bin\\tinyiiod.dll exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\lib\\tinyiiod.lib exit 1  # [win]
```
Comment (carterbox) [inline recipes/libtinyiiod/meta.yaml:33]: Please add an existence test for one of the cmake files.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/libtinyiiod/meta.yaml (hunk 1 above)
Comment (tfcollins) [reply recipes/libtinyiiod/meta.yaml:13]: Updated

File: recipes/libtinyiiod/meta.yaml (hunk 2 above)
Comment (tfcollins) [reply recipes/libtinyiiod/meta.yaml:35]: Updated

**Comment by tfcollins** [comment]
Ready for another review @conda-forge/help-c-cpp

**Pull request merged**